src/prices/prices.controller.ts

```diff
@@ -0,0 +1,33 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { PricesService } from './prices.service';
+import { PricesResponse } from './models/prices.model';
+import { ApiAcceptedResponse, ApiBadRequestResponse } from '@nestjs/swagger';
+import { PricesResponseDto } from './responses/prices.response';
+import { GetForecastedPricesQuery } from './input/get-forecast.query';
+
+@Controller('prices')
+export class PricesController {
+  constructor(private readonly pricesService: PricesService) {}
+
+  @Get('current')
+  @ApiAcceptedResponse({
+    type: PricesResponseDto,
+  })
+  async getCurrentPrices(): Promise<PricesResponse> {
+    return await this.pricesService.getCurrentPrices();
```

Comment:
no need to `return await` here - just `return` should be enough 👍🏻 Then you can also get rid of `async` (do not remove `Promise` though). You would have to use `return await` if you had it wrapped in `try ... catch` block, but here we don't have any.